src/qt/addresstablemodel.cpp

```diff
@@ -452,3 +475,24 @@ void AddressTableModel::emitDataChanged(int idx)
 {
     Q_EMIT dataChanged(index(idx, 0, QModelIndex()), index(idx, columns.length()-1, QModelIndex()));
 }
+
+CPubKey AddressTableModel::getRawPubKey()
+{
+    CPubKey newKey;
+    if(!wallet->GetKeyFromPool(newKey))
+    {
+        WalletModel::UnlockContext ctx(walletModel->requestUnlock());
+        if(!ctx.isValid() || !wallet->GetKeyFromPool(newKey))
+            throw std::runtime_error("Unable to get a new key from keypool.");
+    }
+    return newKey;
+}
+
+QString AddressTableModel::getRawPubKeyString()
+{
+    try {
+        return QString::fromStdString(HexStr(getRawPubKey()));
```

Comment:
why throw a runtime error at all if you will capture it and silence it?

The behavior of this function and the `getRawPubKey()` are not the same in the event you can't get the key from the pool or the unlock context is invalid.

Perhaps it's better to stick to one approach on both of them, either throw the exception on both, or not.